sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -515,15 +515,15 @@ def make_predictor(self, bin_thresholds=None):
         -------
         A TreePredictor object.
         """
-        predictor_nodes = np.zeros(self.n_nodes, dtype=PREDICTOR_RECORD_DTYPE)
+        predictor_nodes = TreePredictor(self.n_nodes)
         _fill_predictor_node_array(predictor_nodes, self.root,
                                    bin_thresholds, self.n_bins_non_missing)
-        return TreePredictor(predictor_nodes)
+        return predictor_nodes
 
 
-def _fill_predictor_node_array(predictor_nodes, grower_node,
-                               bin_thresholds, n_bins_non_missing,
-                               next_free_idx=0):
+def _fill_predictor_node_arry(predictor_nodes, grower_node,
```

Comment:
there seem to be a typo here?

Though it seems this whole function should be removed